lib/compat.php

```diff
@@ -228,6 +228,27 @@ function gutenberg_get_post_type_capabilities( $user, $name, $request ) {
 	return $value;
 }
 
+/**
+ * Make sure oEmbed REST Requests apply the WP Embed security mechanism for WordPress embeds.
+ *
+ * TODO: This is a temporary solution. This code should be included in WordPress Core.
+ *
+ * @since ?
+ *
+ * @param  array $data The REST Request response.
+ * @return array       The REST Request response.
+ */
+function gutenberg_filter_oembed_result( $data ) {
+	if ( ! defined( 'REST_REQUEST' ) || false === REST_REQUEST || empty( $data['html'] ) || empty( $_GET['url'] ) ) {
+		return $data;
+	}
+
+	$data['html'] = wp_filter_oembed_result( $data['html'], (object) $data, $_GET['url'] );
```

Comment:
Minor: This function could be made more concise and have fewer return paths if we just moved this single line into the negation of the previous condition:

```php
<?php

function gutenberg_filter_oembed_result( $data ) {
	if ( defined( 'REST_REQUEST' ) && false !== REST_REQUEST && ! empty( $data['html'] ) && ! empty( $_GET['url'] ) ) {
		$data['html'] = wp_filter_oembed_result( $data['html'], (object) $data, $_GET['url'] );
	}

	return $data;
}
```

Reply:
Great, thanks a lot.